plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcPageSourceFactory.java

```diff
@@ -126,39 +127,49 @@
     private final FileFormatDataSourceStats stats;
     private final DateTimeZone legacyTimeZone;
     private final int domainCompactionThreshold;
+    private final TrinoFileSystemFactory fileSystemFactory;
 
     @Inject
-    public OrcPageSourceFactory(OrcReaderConfig config, HdfsEnvironment hdfsEnvironment, FileFormatDataSourceStats stats, HiveConfig hiveConfig)
+    public OrcPageSourceFactory(
+            OrcReaderConfig config,
+            HdfsEnvironment hdfsEnvironment,
+            FileFormatDataSourceStats stats,
+            HiveConfig hiveConfig,
+            TrinoFileSystemFactory fileSystemFactory)
```

Comment:
This is kinda a half-way migration right? We still need the hdfsEnvironment for some places. Let's put the two fields together in the field list